### PR TITLE
feat: add WASM example

### PR DESCRIPTION
**Pull request by MegaRedHand**
## Description

This PR moves the [cairo-vm-wasm](https://github.com/lambdaclass/cairo-rs-wasm/tree/main) example to this repo and updates it.

We can later improve it, maybe adding an input and passing it to the Cairo program.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [X] CHANGELOG has been updated.

### CODE REVIEW

File: examples/wasm-demo/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+[package]
+name = "wasm-demo"
+description = "A demo using cairo-vm in a WASM environment"
+version.workspace = true
+edition.workspace = true
+license.workspace = true
+repository.workspace = true
+readme.workspace = true
+keywords.workspace = true
```
Comment (Oppen) [inline examples/wasm-demo/Cargo.toml:9]: It may be a good idea to set `publish = false`.

File: examples/wasm-demo/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+[package]
+name = "wasm-demo"
+description = "A demo using cairo-vm in a WASM environment"
+version.workspace = true
+edition.workspace = true
+license.workspace = true
+repository.workspace = true
+readme.workspace = true
```
Comment (Oppen) [inline examples/wasm-demo/Cargo.toml:8]: I see we have a README made specifically for the demo. Are we sure we don't want to point to that one?

File: examples/wasm-demo/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+keywords.workspace = true
+
+[lib]
+crate-type = ["cdylib", "rlib"]
+
+[features]
+default = ["console_error_panic_hook"]
+
+[dependencies]
+serde_json = "1.0.87"
```
Comment (Oppen) [inline examples/wasm-demo/Cargo.toml:18]: Could we use a workspace dep for this?

File: examples/wasm-demo/Cargo.toml (hunk 1 above)
Comment (MegaRedHand) [reply examples/wasm-demo/Cargo.toml:9]: Yeah 😅

**Review by Oppen — APPROVED** [review]

File: examples/wasm-demo/Cargo.toml (hunk 2 above)
Comment (MegaRedHand) [reply examples/wasm-demo/Cargo.toml:8]: You're right! I didn't think about that.

File: examples/wasm-demo/Cargo.toml (hunk 3 above)
Comment (MegaRedHand) [reply examples/wasm-demo/Cargo.toml:18]: Sure!

**Review by fmoletta — APPROVED** [review]

**Comment by Oppen** [comment]
@juanbono I see we're stuck waiting for a "Run tests" job. That doesn't exist, can we remove it? We want "QA / Run tests" instead in all its variants (I see some missing).

**Comment by MegaRedHand** [comment]
We need to remove the _"Run tests (test)"_ required check and add:

- Run tests (test#1)
- Run tests (test#2)
- Run tests (lambdaworks-felt, test#1)
- Run tests (lambdaworks-felt, test#2)
- Run tests (lambdaworks-felt, test-no_std)
- Run tests (lambdaworks-felt, test-wasm)

**Comment by MegaRedHand** [comment]
I'm leaving this here for any future reappearances of this problem (`Unhandled exception. System.IO.IOException: No space left on device`):

Ideally, we would enable some option of `llvm-cov` that doesn't produce intermediate artifacts (`.profraw` files). Alas, I didn't find any way to do this, so I assume it isn't possible.

We can work around this problem by:
- freeing disk space on the GitHub runners via [maximize-build-disk-space action](https://github.com/marketplace/actions/maximize-build-disk-space) (this is a hack),
- dividing the tests into more partitions and running them in parallel,
- dividing the tests into more partitions but running them sequentially, while cleaning `*.profraw` files in-between runs,

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -184,17 +187,16 @@ jobs:
     - name: Check no-std
       run: |
         cd ensure-no_std
-        cargo build --no-default-features
-        cargo build
+        cargo check --no-default-features
```
Comment (pefontana) [inline .github/workflows/rust.yml:190]: I think it is better to run cargo build instead of a check.

**Review by pefontana — APPROVED** [review]

**Pull request merged**